pkg/agent/handler/accessrequest.go

```diff
@@ -82,16 +78,15 @@ func (h *accessRequestHandler) Handle(action proto.Event_Type, _ *proto.EventMet
 
 	var status prov.RequestStatus
 
-	if ar.Status.Level == statusPending && ar.State.Name == provision {
+	if ar.Status.Level == statusPending {
 		status = h.prov.AccessRequestProvision(req)
 	}
 
-	if ar.Status.Level == statusPending && ar.State.Name == deprovision {
+	if ar.Status.Level == statusPending {
```

Comment:
This does not seem right, the pending status is being used for both provisioning and deprovisioning

Reply:
The state is being remove, but this should be updated to check for DELETING state in resource metadata

Reply:
ar.Metadata.State == "DELETING", right ? Just confirming